ruby/lib/minitest/queue/build_status_reporter.rb

```diff
@@ -167,7 +167,11 @@ def report
 
       def success?
         build.error_reports.empty? &&
-          build.worker_errors.empty?
+          build.worker_errors.empty? &&
+          build.queue_exhausted? &&
+          !supervisor.max_test_failed? &&
+          !all_workers_died? &&
```

Comment:
The only missing case should've been `all_workers_died` because the other cases are implicitly checked through `error_reports`. I've added them nevertheless to be 100% sure though.